pages/studentPortal/web/React/assignments/template.tsx

```diff
@@ -11,44 +11,72 @@ import { Button } from "../../../../../components/ui/Button";
 import {
   FETCH_USER_ASSIGNMENT_SUBMISSIONS,
   FetchUserAssignmentSubmissionsDataResponse,
-  UserAssignmentSubmissionsData,
 } from "../../../../../graphql/fetchUserAssignmentSubmissions";
 import { useAuth } from "../../../../../lib/authContext";
+import {
+  assignmentsSelector,
+  setUserAssignments,
+} from "../../../../../redux/assignmentsSlice";
+import { selectionSetMatchesResult } from "@apollo/client/cache/inmemory/helpers";
 
 const React2 = ({ incompleteStage, submittedStage, completedStage }) => {
   const router = useRouter();
   const [stage, setStage] = useState(0);
   const { user } = useAuth();
-  const [assignments, setAssignments] = useState<
-    UserAssignmentSubmissionsData[]
-  >([]);
+  const { userAssignments } = useSelector(assignmentsSelector);
+  const dispatch = useDispatch();
+
   // REQUIRED: create assignment in coding_assignments table to generate ID and paste here
   const assignmentId = "2cf9156a-4f6f-452d-b09a-2c54f19a7b40";
 
-  const { data } = useQuery<FetchUserAssignmentSubmissionsDataResponse>(
-    FETCH_USER_ASSIGNMENT_SUBMISSIONS,
-    {
-      variables: {
-        user_id: user.uid,
-        assignmentId: assignmentId,
-      },
+  useEffect(() => {
+    if (userAssignments) {
+      const currentAssignment = userAssignments.filter(
+        (assignment) => assignment.assignmentId === assignmentId
+      );
+      let assignmentReviewLink = currentAssignment.map(
+        (assignment) => assignment.review_link
+      );
+      let assignmentSubmissionLink = currentAssignment.map(
+        (assignment) => assignment.submission_link
+      );
+      if (assignmentReviewLink != null) {
```

Comment:
can you just refer to currentAssignment.review_link and currentAssignment.submission_link?

Reply:
I can if I change the filter I used on line 34 to find... or do the first record..

I changed to this:
 const currentAssignment = userAssignments.find(
        (assignment) => assignment.assignmentId === assignmentId
      );

and then I can access currentUserAssignment.review_link etc directly.